pkg/agent/config/node_config.go

```diff
@@ -59,6 +59,13 @@ var (
 	VirtualServiceIPv6 = net.ParseIP("fc01::aabb:ccdd:eeff")
 )
 
+type NodeType string
```

Comment:
If we often compare the values, probably use enum.

Reply:
With feature gate, this is not ofen used in value comparasion. I want to keep string value because I planed to show the Node type in AntreaAgentInfo in future.

Reply:
feature gate is not used in "inner" pkgs. Always converted to a boolean.

I mean if we compare the values in any frequent call path (e.g. when handling a new change from API or CNI), I think enum is better. If it is just used in bootstrapping, string is fine. Even for enum, you can add a String() func.

Reply:
feature gate is not used in "inner" pkgs. Always converted to a boolean.

I mean if we compare the values in any frequent call path (e.g. when handling a new change from API or CNI), I think enum is better. If it is just used in bootstrapping, string is fine. Even for enum, you can add a String() func.